Content.Server/Power/EntitySystems/PowerNetSystem.cs

```diff
@@ -86,6 +89,11 @@ private void ApcPowerReceiverRemove(EntityUid uid, ApcPowerReceiverComponent com
             component.NetworkLoad.Paused = false;
         }
 
+        private void ApcPowerReceiverPowerChanged(Entity<AppearanceComponent> ent, ref PowerChangedEvent args)
```

Comment:
now that his is here, dont you need to delete the SetData in ProviderChanged?

Reply:
Moved back to the loop. Also removed the redundant event in ProviderChanged.